server/core/src/https/errors.rs

```diff
@@ -0,0 +1,69 @@
+//! Where we hide the error handling widgets
+//!
+
+use axum::response::{IntoResponse, Response};
+use http::header::ACCESS_CONTROL_ALLOW_ORIGIN;
+use http::{HeaderValue, StatusCode};
+use kanidm_proto::v1::OperationError;
+use utoipa::ToSchema;
+
+/// The web app's top level error type, this takes an `OperationError` and converts it into a HTTP response.
+#[derive(Debug, ToSchema)]
+pub enum WebError {
+    /// Something went wrong when doing things.
+    OperationError(OperationError),
+    InternalServerError(String),
+}
+
+impl From<OperationError> for WebError {
```

Comment:
Is this kind of tied into the error handling stuff we want to do in future?

Reply:
Yep, it's a look at how that kind of thing could work in the future.